### DISCUSSION TITLE
Questions about exporting XRD datas

### DISCUSSION THREAD

**Discussion by Laurah11** [mtex-toolbox/mtex#1856] [Ask Anything] [answered]
Dear all, 

I am currently working on XRD datas and I want to export them to be used in other codes or softwares (VPSC for instance). 

I have several questions about the way it is done on MTEX : 

1. What is the difference between doing : 

```matlab
S3G = equispacedSO3Grid(CS,SS,'resolution',5*degree);
export(pdf,'name.txt',S3G,'Bunge','generic')
``` 
or 
```matlab
odf.export('name_VPSC.txt','VPSC')
``` 
In terms of the grid formation ? Because, from my understanding both should have a discretisation of the Euler Space of 5° resolution. 
The VPSC file is well every 5° for the three angles but for the first method, it seems more chaotic and depend of the initial ODF. 
Is it based on the difference between using  ```regularSO3Grid ``` or  ```equispacedSO3Grid``` ? If so could you please explain a little bit more the difference between the two ? 

2. When creating a grid thanks to ``` equispacedSO3Grid```, ```regularSO3Grid ```  or the VPSC interface, does a volume fraction of the mean orienation is calculated in the corresponding subspace or only the value of the ODF at the specific point in the Euler space ? 

3. When we use``` DiscreteSample(odf, nb points) ```  or ```export_VPSC(odf, 'name.txt,'points',10000)```  : are the orientation chosen purely random or does it take into account the initiale density of orientation ? For instance if we have a strong Cube texture,  does it take into account that and it leads that it will be more probable to have orientation around this component ? 

Thanks you in advance and for any help you can provide. For any additional info, do not hesitate to ask.

**Comment by kilir** [comment, marked as answer]
Hi,
wrt 1) assuming in your first example `pdf` is an odf the difference is, that the VPSC format takes randomly drawn orientations from an odf while `export_generic` evaluates the odf at grid points and exports those points and the corresponding values.

wrt 2) the values of the odf is evaluated just at that point

wrt 3) orientations are drawn from the odf such that orientations within volumes of higher densities are more probable to be drawn.

Hope this helps,

Cheers,
Rüdiger

**Reply by Laurah11** [reply]
Hi, 

Thank you very much for the answers. It helps a lot !

Just a last question : when creating the grid point, can you explain to me the difference between using ```regularSO3Grid```  or ``` equispacedSO3Grid```  ?

Thanks again ! 

Laura

**Reply by ralfHielscher** [reply]
Look at the difference between the regular and equispaced S2Grid
<img width="278" alt="reg" src="https://github.com/mtex-toolbox/mtex/assets/8619241/b4796d11-d69f-4c5d-94e9-6be3f174a884">
<img width="279" alt="equi" src="https://github.com/mtex-toolbox/mtex/assets/8619241/26dfeb99-36af-40ec-b22c-7aa40f03f173">
The difference for the SO3Grid is the same. You should not use the regular SO3Grid for vpsc.

Ralf.

**Reply by Laurah11** [reply]
Hi, 

I see better now. 
Thank you very much for your help ! 

Laura
